mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp

```diff
@@ -335,7 +412,7 @@ static void printNamedStructuredOp(OpAsmPrinter &p, Operation *op,
       /*elidedAttrs=*/{"operandSegmentSizes",
                        // See generated code in
                        // LinalgNamedStructuredOps.yamlgen.cpp.inc
-                       "linalg.memoized_indexing_maps"});
+                       "linalg.memoized_indexing_maps", "indexing_maps"});
```

Comment:
I'm trying to decide whether this is a good idea or not to have as one list for every op vs pushing to the caller.

If we fast forward and introduce an op that wants its indexing maps atte printed as a default attrdict attr, then we're going to have to unwind what this was in support of. 

Maybe just move elidedAttrs to an argument and pass it explicitly from each call site. Then it will be easy to find exactly what is being done. I would move the operandSegmentSizes to the call site too in order to avoid splicing the list here.

Reply:
This seems to be reminiscent of the OpDSL mechanism. Changing the caller side may not be trivial without changing OpDSL itself, since the only call to this method (currently) is from inside `mlir-linalg-ods-yaml-gen.cpp`, and now from `matmul` in this PR.

We could just change the YAML generator with some default arguments, but we could also move the three matmuls and then refactor? We could also not reuse the same function, duplicate some of it elsewhere and leave the "NamedStructuredOp" stuff for OpDSL only.

I don't mind either way.

Reply:
It's fine to defer if you all are going to keep working on this and get it to a final state. I guess I'm less scared of making a single line change to mlir-linalg-ods-yaml-gen.cpp to move the callsite but I understand if folks don't want to do that.

Reply:
I have incorporated the requested change. Hope my understanding was correct.